Added the setting to protect the /metrics endpoint

henriqueccapozzi opened the pull request:
Modified the 'ExportToDjangoView' so that when the variables

```
DJANGO_PROMETHEUS_AUTHORIZATION_USERNAME
DJANGO_PROMETHEUS_AUTHORIZATION_PASSWORD
```

are declared on `settings.py` and have a variable different than `None`

Metrics will only be retrieved when the request have basic authentication set
to the expected values

asherf commented:
Hi @henriqueccapozzi thanks for the PR.
However, I don't think this is a good idea. 
using the ExportToDjangoView & url defined in this package are completely optional.
If one has specific requirements around the view used to export prometheus metrics, it is trivial to create your own view in your app and implement whatever auth/authz logic in that view.
For the view provided by this package, I think it is useful to keep things simple.
one can even implement their own auth logic in a view and if the auth logic passes just delegate to `ExportToDjangoView`
for example:

```python

def prometheus_metrics_view(request):
    if not check_auth(request):
       return HttpResponseForbidden()
    return ExportToDjangoView(request)
```

henriqueccapozzi commented:
That's an even better idea than what I did
Thanks very much for the direction

Following what you said, i would hook this custom view to the /metrics url and implement the auth on it
Roght?
@asherf

asherf commented:
> That's an even better idea than what I did
> Thanks very much for the direction
> 
> Following what you said, i would hook this custom view to the /metrics url and implement the auth on it
> Roght?
> @asherf

yep.

henriqueccapozzi commented:
> > That's an even better idea than what I did
> > Thanks very much for the direction
> > Following what you said, i would hook this custom view to the /metrics url and implement the auth on it
> > Roght?
> > @asherf
> 
> yep.

Thanks very much 
It worked very nicely



For future reference or if anyone wants to implement it, that's how I did it


```python
# settings.py
DJANGO_PROMETHEUS_AUTHORIZATION_USERNAME = 'user'
DJANGO_PROMETHEUS_AUTHORIZATION_PASSWORD = 'password'

# urls.py
from .views import prometheus_metrics_view
urlpatterns = [
   # ...
   path('metrics', prometheus_metrics_view, name="django-prometheus-view"),
   # ....
]

# view.py
import base64

from django.conf import settings
from django.http import HttpResponse
from django_prometheus.exports import ExportToDjangoView


def prometheus_metrics_view(request):
    expected_username = getattr(
        settings, "DJANGO_PROMETHEUS_AUTHORIZATION_USERNAME", None)
    expected_password = getattr(
        settings, "DJANGO_PROMETHEUS_AUTHORIZATION_PASSWORD", None)
    if expected_password is not None and expected_username is not None:
        auth_header = request.META.get("HTTP_AUTHORIZATION", "")
        token_type, _, credentials = auth_header.partition(" ")
        if credentials == '':
            return HttpResponse("", status=400)

        received_auth_string = base64.b64decode(credentials).decode()
        if ':' not in received_auth_string:
            return HttpResponse("", status=400)

        received_username = received_auth_string.split(':')[0]
        received_password = received_auth_string.split(':')[1]

        valid_username = received_username == expected_username
        valid_password = received_password == expected_password

        if token_type != 'Basic' or not valid_username or not valid_password:
            return HttpResponse("", status=401)

    return ExportToDjangoView(request)

```
